### ISSUE TITLE
Bug Report 2022/01/05

### ISSUE THREAD

**Issue by acjvenis** [magento/magento2-phpstorm-plugin#908]
**Describe the bug** (*)

Introducing class preference in di.xml file

```
java.lang.IllegalArgumentException: URLDecoder: Incomplete trailing escape (%) pattern
	at java.base/java.net.URLDecoder.decode(URLDecoder.java:225)
	at com.magento.idea.magento2plugin.project.diagnostic.github.GitHubNewIssueBodyBuilderUtil.decode(GitHubNewIssueBodyBuilderUtil.java:133)
	at com.magento.idea.magento2plugin.project.diagnostic.github.GitHubNewIssueBodyBuilderUtil.buildNewBugReportBody(GitHubNewIssueBodyBuilderUtil.java:60)
	at com.magento.idea.magento2plugin.project.diagnostic.github.GitHubNewIssueUrlBuilderUtil.buildNewBugIssueUrl(GitHubNewIssueUrlBuilderUtil.java:39)
	at com.magento.idea.magento2plugin.project.diagnostic.DefaultErrorReportSubmitter.submit(DefaultErrorReportSubmitter.java:73)
	at com.intellij.diagnostic.IdeErrorsDialog.reportMessage(IdeErrorsDialog.java:636)
	at com.intellij.diagnostic.IdeErrorsDialog.reportAll(IdeErrorsDialog.java:1103)
	at com.intellij.diagnostic.IdeErrorsDialog$ReportAndClearAllAction.actionPerformed(IdeErrorsDialog.java:1087)
	at com.intellij.diagnostic.IdeErrorsDialog$CompositeAction.actionPerformed(IdeErrorsDialog.java:994)
	at java.desktop/javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1967)
	at java.desktop/javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2308)
	at java.desktop/javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:405)
	at java.desktop/javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:262)
	at java.desktop/javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:270)
	at java.desktop/java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:297)
	at java.desktop/java.awt.Component.processMouseEvent(Component.java:6652)
	at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3345)
	at java.desktop/java.awt.Component.processEvent(Component.java:6417)
	at java.desktop/java.awt.Container.processEvent(Container.java:2263)
	at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5027)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2321)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4859)
	at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4918)
	at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4547)
	at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4488)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2307)
	at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2784)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4859)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:778)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:95)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:751)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:749)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:748)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:885)
	at com.intellij.ide.IdeEventQueue.dispatchMouseEvent(IdeEventQueue.java:814)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:751)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$6(IdeEventQueue.java:441)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:825)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$7(IdeEventQueue.java:440)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:794)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:492)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)

```

**To Reproduce** (*)

**Expected behavior** (*)

**Screenshots**

**Please complete the following information:** (*)

 - OS: Mac OS X 10.15.7
 - PhpStorm/Intellij version: 2021.2.3
 - Plugin Version: 4.2.1

**Additional context**

**Comment by Iamwade** [comment]
@magento I am working on this

**Comment by Iamwade** [comment]
Hello, @acjvenis!
Thank you for your report!
This issue has been already resolved in the scope of #842 number.
It will be available in the next release: 4.2.2.
Regards

**Comment by bohdan-harniuk** [comment]
Hello, @Iamwade!

Thank you for your investigation!
